third-party/realdds/include/realdds/dds-device-impl.h

```diff
@@ -275,48 +245,44 @@ class dds_device::impl
                             if( state_type::WAIT_FOR_PROFILES == state )
                             {
                                 LOG_INFO( "got MOTION_STREAM_PROFILES message" );
-                                auto motion_stream_profiles
-                                    = data.get< topics::device::notification::motion_stream_profiles_msg >();
-
-                                // TODO: Try to save the "emplace" profiles copy
-                                topics::device::notification::motion_stream_profiles_msg msg;
-                                msg.dds_sensor_index = motion_stream_profiles->dds_sensor_index;
-                                msg.num_of_profiles = motion_stream_profiles->num_of_profiles;
-                                for( int i = 0; i < motion_stream_profiles->num_of_profiles; ++i )
-                                    msg.profiles[i] = motion_stream_profiles->profiles[i];
-
-                                auto sensor_name_it
-                                    = _sensor_index_to_name.find( motion_stream_profiles->dds_sensor_index );
-                                if( sensor_name_it != _sensor_index_to_name.end() )
+
+                                auto profiles_msg = data.get< topics::device::notification::motion_stream_profiles_msg >();
```

Comment:
I know it wasn't there, but maybe we need to add some protection for not accessing null?
for all places accessing the return data from `get<>`
If the server sends junk the app will crash

Reply:
Done